Mathlib/Algebra/DirectSum/Basic.lean

```diff
@@ -170,10 +170,10 @@ theorem of_injective (i : ι) : Function.Injective (of β i) :=
   DFinsupp.single_injective
 
 @[elab_as_elim]
-protected theorem induction_on {C : (⨁ i, β i) → Prop} (x : ⨁ i, β i) (H_zero : C 0)
-    (H_basic : ∀ (i : ι) (x : β i), C (of β i x))
-    (H_plus : ∀ x y, C x → C y → C (x + y)) : C x := by
-  apply DFinsupp.induction x H_zero
+protected theorem induction_on {C : (⨁ i, β i) → Prop} (x : ⨁ i, β i) (zero : C 0)
```

Comment:
You can declare it out of scope for this PR, but the naming guide says
```suggestion
protected theorem induction_on {motive : (⨁ i, β i) → Prop} (x : ⨁ i, β i) (zero : motive 0)
```
and since you're modifying these lines anyway...